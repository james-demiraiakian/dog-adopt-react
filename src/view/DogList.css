.dog-list {
  display: flex;
  margin-top: 15vw;
  padding-left: 100%;
  animation: float 30s infinite linear;
  white-space: nowrap;
}

.dog-list:hover {
  animation-play-state: paused;
}

@keyframes float {
  to {
    transform: translateX(-100%);
  }
}

.dog-link {
  margin-left: 1vw;
  /* border: 5px solid black;
  border-radius: 20px;
  box-shadow: 10 10 10 black; */
}
