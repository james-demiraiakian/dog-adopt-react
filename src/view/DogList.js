import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getDogs } from '../services/dogs';

export default function DogList() {
  const [dogs, setDogs] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await getDogs();
      setDogs(data);
      setLoading(false);
      console.log(data);
    };
    fetchData();
  }, []);

  return (
    <div className="dog-list">
      {dogs.map((d) => (
        <Link key={d.id} to={`/dogs/${d.id}`}>
          {d.name}
        </Link>
      ))}
    </div>
  );
}
