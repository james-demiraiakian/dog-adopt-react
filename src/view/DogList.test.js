import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import DogList from './DogList';

test('should render a list of dogs', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <Route path="/" component={DogList} />
    </MemoryRouter>
  );

  await screen.findByText('Eyes', 'Althea');

  expect(container).toMatchSnapshot();
});
