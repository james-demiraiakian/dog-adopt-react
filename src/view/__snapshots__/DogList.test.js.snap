// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of dogs 1`] = `
<div>
  <div
    class="dog-list"
  >
    <a
      class="dog-link"
      href="/dogs/2"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=12"
        />
        <h3>
          Amely
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/3"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=10"
        />
        <h3>
          Althea
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/4"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=51"
        />
        <h3>
          Franny
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/5"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=8"
        />
        <h3>
          Melba
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/6"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://images.pexels.com/photos/6246667/pexels-photo-6246667.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260"
        />
        <h3>
          Ron
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/8"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=4"
        />
        <h3>
          Eileen
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/9"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=5"
        />
        <h3>
          Liliana
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/10"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=6"
        />
        <h3>
          Barton
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/11"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=21"
        />
        <h3>
          Happy
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/14"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=22"
        />
        <h3>
          Eyes
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/15"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/500?id=26"
        />
        <h3>
          Pugilicious
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/16"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://images.pexels.com/photos/5886866/pexels-photo-5886866.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
        />
        <h3>
          Ginny
        </h3>
      </div>
    </a>
    <a
      class="dog-link"
      href="/dogs/17"
    >
      <div
        class="dog-crate"
      >
        <img
          class="image"
          src="https://placedog.net/1000/563?id=3"
        />
        <h3>
          Mrs. Weasley
        </h3>
      </div>
    </a>
  </div>
</div>
`;
